### PR TITLE
[pytx] CLI separation: move config and state classes out of cli module

### PR DESCRIPTION
**Pull request by dougneal**
Summary
---------

The library's core functionality and the CLI are separate concerns, so code which doesn't concern the CLI, shouldn't have the appearance of doing so. Move and rename the config and state management classes to reflect this. This will be fundamental to making the API parts make sense.

Test Plan
---------

See what GH CI says...

### CODE REVIEW

File: python-threatexchange/threatexchange/state.py [hunk 1]
```diff
@@ -148,12 +153,12 @@ def _read_state(
...
             return delta
-        except Exception:
+        except Exception as exc:
             logging.exception("Failed to read state for %s", collab_name)
-            raise CommandError(
+            raise TXStateException(
                 f"Failed to read state for {collab_name}. "
                 "You might have to delete it with `threatexchange fetch --clear`"
-            )
+            ) from exc
```
Comment (Dcallies) [inline python-threatexchange/threatexchange/state.py:161]: I think this is captured by default in modern python, and might explicitly hint you don't want the full stack printed (or maybe that's `from None`

File: python-threatexchange/threatexchange/state.py [hunk 2]
```diff
@@ -32,9 +31,15 @@
 from threatexchange.signal_type import index
 
 
-class CliIndexStore:
+class TXStateException(Exception):
```
Comment (Dcallies) [inline python-threatexchange/threatexchange/state.py:34]: This is going to collide with Nikolay's changes if he has any in flight,

File: python-threatexchange/threatexchange/state.py (hunk 1 above)
Comment (dougneal) [reply python-threatexchange/threatexchange/state.py:161]: Oh ok, I was following a linter recommendation but it may have been an out of date rule

**Comment by dougneal** [comment]
> I think it might be better at this exact time to just import the CLI libraries as they are named - this is something that is easy for us to do after the hackathon, but may be disruptive during the hackathon.
> 
> That, plus the other considerations on whether we want to promote these interfaces fully make this not clear cut to me.
> 
> Is there anything being blocked by these changes that requires them?

No prob. I'll close this.

**Pull request closed without merging**